src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesV2Spec.scala

```diff
@@ -1,115 +1,156 @@
+import org.broadinstitute.dsde.workbench.model.ErrorReportJsonSupport._
+import org.broadinstitute.dsde.workbench.sam.matchers.BeForSamUserResponseMatcher.beForUser
 import org.broadinstitute.dsde.workbench.sam.model._
-import org.broadinstitute.dsde.workbench.sam.service.UserService._
-import org.broadinstitute.dsde.workbench.sam.service.{NoExtensions, StatusService, TosService, UserService}
-
-/** Created by mtalbott on 8/8/18.
-  */
-class UserRoutesV2Spec extends UserRoutesSpecHelper {
-  def withSARoutes[T](testCode: (TestSamRoutes, TestSamRoutes) => T): T = {
-    val directoryDAO = new MockDirectoryDAO()
-
-    val tosService = new TosService(directoryDAO, TestSupport.tosConfig)
-    val samRoutes = new TestSamRoutes(
-      null,
-      null,
-      new UserService(directoryDAO, NoExtensions, Seq.empty, tosService),
-      new StatusService(directoryDAO, NoExtensions),
-      null,
-      defaultUser,
-      NoExtensions,
-      tosService = tosService
-    )
-    val SARoutes = new TestSamRoutes(
-      null,
-      null,
-      new UserService(directoryDAO, NoExtensions, Seq.empty, tosService),
-      new StatusService(directoryDAO, NoExtensions),
-      null,
-      petSAUser,
-      NoExtensions,
-      tosService = tosService
-    )
-    testCode(samRoutes, SARoutes)
-  }
+import org.broadinstitute.dsde.workbench.sam.model.api.{SamUser, SamUserAllowances, SamUserResponse}
+import org.broadinstitute.dsde.workbench.sam.service._
+import org.broadinstitute.dsde.workbench.sam.{Generator, TestSupport}
+import org.mockito.scalatest.MockitoSugar
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
 
-  "POST /register/user/v2/self" should "create user" in withDefaultRoutes { samRoutes =>
-    Post("/register/user/v2/self") ~> samRoutes.route ~> check {
-      status shouldEqual StatusCodes.Created
-      val res = responseAs[UserStatus]
-      res.userInfo.userSubjectId.value.length shouldBe 21
-      res.userInfo.userEmail shouldBe defaultUserEmail
-      res.enabled shouldBe TestSupport.enabledMapNoTosAccepted
-    }
+class UserRoutesV2Spec extends AnyFlatSpec with Matchers with ScalatestRouteTest with MockitoSugar with TestSupport {
+  val defaultUser: SamUser = Generator.genWorkbenchUserGoogle.sample.get
+  val otherUser: SamUser = Generator.genWorkbenchUserGoogle.sample.get
+  val adminGroupEmail: WorkbenchEmail = Generator.genFirecloudEmail.sample.get
+  val allUsersGroup: BasicWorkbenchGroup = BasicWorkbenchGroup(CloudExtensions.allUsersGroupName, Set(), WorkbenchEmail("all_users@fake.com"))
+
+  "GET /api/users/v2/self" should "get the user object of the requesting user" in {
+    // Arrange
+    val samRoutes = new MockSamRoutesBuilder(allUsersGroup)
+      .withEnabledUser(defaultUser) // "persisted/enabled" user we will check the status of
+      .withAllowedUser(defaultUser) // "allowed" user we will check the status of
```

Comment:
Since these tests are on the API routing layer, they should not have a distinction between and allowed and enabled user. I think we should defer to using `withAllowedUser`

Reply:
If not possible otherwise, this may be a place where we need to mock out the only routes for backwards compatibility.

Reply:
I wasn't sure what to do here. I think it is valuable, because a user being enabled doesn't necessarily mean they can use Terra. I think there's value in being able to test and respond accordingly. I do agree that this is clunky, having to specify _both_ things. Maybe all usages of `withEnabledUser(s)` should be migrated to a new `withReadyUser(s)` call, which would do both enabling and allowing, leaving the other two methods there for testing the specifics if need be.

Reply:
Good point. I'll change things.

Reply:
Actually, I'm going to take it back. As I was rewriting, I realized that the difference between `enabled` and `allowed` _is_ significant to the api layer. They mean different things. `allowed` for the v2 endpoints, means the user is able to use the system, which means they're both `enabled` in the db _AND_ have accepted the Terms of Service. These things are spelled out in the API responses, so we need to be able to test them. The test code as written doesn't actually mandate that the test "know" that `allowed` means `enabled` and ToS accepted, but the responses should look right.

Reply:
It looks like this has been modified, but IIRC I was trying to have these things call each other so that if I wanted to say `.withEnabledUser` it would take care of creating the user automatically.  So carrying that logic forward, I would expect `.withAllowedUser` to make the user enabled as well, since for a user to be Allowed it must also be enabled.